client/lorawan.go

```diff
@@ -0,0 +1,47 @@
+package client
+
+import (
+	"context"
+	"io"
+	"net/http"
+)
+
+// Client used to communicate to external services
+type Client interface {
+	Do(*http.Request) (resp *http.Response, err error)
+}
+
+// LoraWAN used to communicate to LoRaWAN external system
+type LoraWAN struct {
```

Comment:
Could this client package/struct encapsulate more information about the API endpoints and their request/response?

For example, rather than having the `LoraWan.DoPost` method, what if we had something that looked more like:

```go
func (l *LoraWAN) OnboardSensor(req  OnboardSensorRequest) (*OnboardSensorResponse, error)
```

This would mean that the specifics of marshalling/unmarshalling the request/response would be contained within this package and wouldn't be leaking out into the `processor` package. If this endpoint were to change slightly, an engineer would have to make changes in both this package and in the `processor` package.